drivers/network/ndis/ndis/miniport.c

```diff
@@ -2399,6 +2399,18 @@ NdisIDispatchPnp(
         Irp->IoStatus.Information |= Adapter->NdisMiniportBlock.PnPFlags;
         break;
 
+      case IRP_MN_REMOVE_DEVICE:
+        IoSkipCurrentIrpStackLocation(Irp);
+        Status = IoCallDriver(Adapter->NdisMiniportBlock.NextDeviceObject, Irp);
+        
+        IoDetachDevice(Adapter->NdisMiniportBlock.NextDeviceObject);
+        
+        RtlFreeUnicodeString(&Adapter->NdisMiniportBlock.SymbolicLinkName);
+        RtlFreeUnicodeString(&Adapter->NdisMiniportBlock.MiniportName);
+        
+        IoDeleteDevice(DeviceObject);
+        return Status;
+
       default:
```

Comment:
Could adding a `NDIS_DbgPrint()` here help notice potential other cases? Or is that not needed/wanted?

Reply:
Not needed. Only the yet-unhandled but actually necessary ones might profit from a dprint tracing.

Reply:
Probably not, but silent default does not help.